### PR TITLE
[Enhancement]: Add issue and PR template

### PR DESCRIPTION
**Pull request by ZwwWayne**
This PR add issue and PR template for MMCV.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/unexpected_report.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+1. Please run `python -c "from mmcv.utils import collect_env; print(collect_env())"` to collect necessary environment information and paste it here.
+2. You may add addition that may be helpful for locating the problem, such as
+    - How you installed PyTorch [e.g., pip, conda, source]
+    - Other environment variables that may be related (such as `$PATH`, `$LD_LIBRARY_PATH`, `$PYTHONPATH`, etc.)
+
+**Error traceback**
+If applicable, paste the error trackback here.
+
+```none
+A placeholder for trackback.
```
Comment (innerlee) [inline .github/ISSUE_TEMPLATE/unexpected_report.md:43]: traceback

File: .github/ISSUE_TEMPLATE/config.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+blank_issues_enabled: false
+
+contact_links:
+  - name: Common Issues
+    url: https://mmdetection.readthedocs.io/en/latest/faq.html
+    about: Check if your issue already has solutions
+  - name: MMDetection Documentation
```
Comment (innerlee) [inline .github/ISSUE_TEMPLATE/config.yml:7]: MMDetection -> mmcv

File: .github/ISSUE_TEMPLATE/config.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+    url: https://mmdetection.readthedocs.io/en/latest/
```
Comment (innerlee) [inline .github/ISSUE_TEMPLATE/config.yml:8]: mmcv

File: .github/ISSUE_TEMPLATE/unexpected_report.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+3. The unexpected results still exist in the latest version.
+
+**Describe the Issue**
+A clear and concise description of what the bug is, including what results are expected and what the real results you got.
+
+**Reproduction**
+
+1. What command, code, or script did you run?
+
+```none
```
Comment (innerlee) [inline .github/ISSUE_TEMPLATE/unexpected_report.md:25]: none -> bash

File: .github/ISSUE_TEMPLATE/unexpected_report.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+A placeholder for the command.
+```
+
+2. Did you make any modifications on the code? Did you understand what you have modified?
+3. What project (e.g, MMDetection, MMAction2) did you use?
```
Comment (innerlee) [inline .github/ISSUE_TEMPLATE/unexpected_report.md:30]: Delete this question?

File: .github/ISSUE_TEMPLATE/general_questions.md [hunk 6]
```diff
@@ -0,0 +1,8 @@
+---
+name: General questions
+about: Ask general questions to get help
+title: ''
+labels: ''
+assignees: ''
+
```
Comment (dreamerlin) [inline .github/ISSUE_TEMPLATE/general_questions.md:7]: **Checklist**

1. I have searched related issues but cannot get the expected help.
2. I have read the [FAQ documentation](https://mmcv.readthedocs.io/en/latest/trouble_shooting.html) but cannot get the expected help.

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
+Thanks for your contribution and we appreciate it a lot.
+
+## Motivation
+Please describe the motivation of this PR and the goal you want to achieve through this PR.
+
+## Modification
+Please briefly describe what modification is made in this PR.
+
+## BC-breaking
+Does the modification introduces changes that breaks the back-compatibility of the downstream repos?
```
Comment (dreamerlin) [inline .github/pull_request_template.md:10]: breaks -> break

File: .github/ISSUE_TEMPLATE/config.yml (hunk 2 above)
Comment (dreamerlin) [reply .github/ISSUE_TEMPLATE/config.yml:7]: It is still `MMDetection`

File: .github/ISSUE_TEMPLATE/config.yml [hunk 8]
```diff
@@ -0,0 +1,9 @@
+blank_issues_enabled: false
+
+contact_links:
+  - name: Common Issues
+    url: https://mmcv.readthedocs.io/en/latest/trouble_shooting.html
+    about: Check if your issue already has solutions
+  - name: MMDetection Documentation
```
Comment (dreamerlin) [inline .github/ISSUE_TEMPLATE/config.yml:7]: It is still MMDetection

File: .github/pull_request_template.md [hunk 9]
```diff
@@ -0,0 +1,21 @@
+Thanks for your contribution and we appreciate it a lot.
+
+## Motivation
+Please describe the motivation of this PR and the goal you want to achieve through this PR.
+
+## Modification
+Please briefly describe what modification is made in this PR.
+
+## BC-breaking
+Does the modification introduces changes that break the back-compatibility of the downstream repos?
```
Comment (dreamerlin) [inline .github/pull_request_template.md:10]: changes -> change

File: .github/ISSUE_TEMPLATE/config.yml (hunk 8 above)
Comment (hellock) [inline .github/ISSUE_TEMPLATE/config.yml:7]: MMCV

File: .github/pull_request_template.md [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.
+
+## Use cases
+If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.
+
+## Checklist
+
+1. Pre-commit or other linting tools are used to fix the potential lint issues.
+2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
+3. If the modification has potential influence on downstream projects, this PR is tested with downstream projects, like MMDet or MMCls.
```
Comment (dreamerlin) [inline .github/pull_request_template.md:20]: this PR is tested with downstream projects -> this PR should be tested with downstream projects

File: mmcv/ops/csrc/onnxruntime/cpu/roi_align_rotated.cpp [hunk 11]
```diff
@@ -2,6 +2,7 @@
 // https://github.com/facebookresearch/detectron2/tree/master/detectron2/layers/csrc/ROIAlignRotated
 // Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
 #include "roi_align_rotated.h"
+
```
Comment (innerlee) [inline mmcv/ops/csrc/onnxruntime/cpu/roi_align_rotated.cpp:5]: unintended change

**Review by dreamerlin — CHANGES_REQUESTED** [review]

File: .github/pull_request_template.md [hunk 12]
```diff
@@ -0,0 +1,21 @@
+Thanks for your contribution and we appreciate it a lot.
```
Comment (hellock) [inline .github/pull_request_template.md:1]: This template is quite complex and may discourage contributions. We can just keep necessary items.

File: mmcv/ops/csrc/onnxruntime/cpu/roi_align_rotated.cpp (hunk 11 above)
Comment (hellock) [inline mmcv/ops/csrc/onnxruntime/cpu/roi_align_rotated.cpp:5]: This modification may not be necessary in this PR.

Comment (ZwwWayne) [reply mmcv/ops/csrc/onnxruntime/cpu/roi_align_rotated.cpp:5]: This is simply to fix the lint issue in the master branch.

File: .github/pull_request_template.md (hunk 9 above)
Comment (ZwwWayne) [reply .github/pull_request_template.md:10]: changes break, or change breaks?

File: .github/pull_request_template.md (hunk 12 above)
Comment (ZwwWayne) [reply .github/pull_request_template.md:1]: I add optional in the title, is it OK?

File: mmcv/ops/csrc/onnxruntime/cpu/roi_align_rotated.cpp (hunk 11 above)
Comment (ZwwWayne) [reply mmcv/ops/csrc/onnxruntime/cpu/roi_align_rotated.cpp:5]: Simply fix the lint error in master branch

**Comment by ZwwWayne** [comment]
Kindly ping @hellock @innerlee @dreamerlin

File: .github/pull_request_template.md (hunk 12 above)
Comment (innerlee) [reply .github/pull_request_template.md:1]: Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

**Review by innerlee — APPROVED** [review]

**Pull request merged**